cmd/coder/auth.go

```diff
@@ -11,15 +11,15 @@ import (
 )
 
 // requireAuth exits the process with a nonzero exit code if the user is not authenticated to make requests
-func requireAuth() *entclient.Client {
-	client, err := newClient()
+func requireAuth(user string) *entclient.Client {
```

Comment:
How come you added the user to create the secret for to the struct? Wouldn't the API be better if it was just a param to the create secret function? Adding the user to the struct implies it's a sort of sudo thing like gitlab.

Reply:
My understanding is that we'll be expanding this concept to nearly all resources in the platform such that an admin could create/start/stop and environment for a specific user. So my thinking is that we'd want to parameterize the target user at the API client abstraction level.

Reply:
My issue with this however is that users of the API will find it tedious to have to recreate a new entclient every time if they wanted to do a Go-based solution for creating a secret for every user. You'd have to make an admin client for getting the user list, then a new client for every user.

How do other APIs handle sudo? I would've thought most APIs would do it like:
```go
type RequestOptions struct {
  Sudo string
}

var DefaultRequestOptions = &RequestOptions{
  Sudo: "me",
}

func (c *Client) GetSomething(ctx context.Context, opts *RequestOptions) {
  if (opts == nil) {
    opts = DefaultRequestOptions
  }

  ...
}
```

Reply:
Ah, this is a good point. Changing now.